x/vendorinfo/types/messages_vendor_info_test.go

```diff
@@ -348,6 +324,28 @@ func TestMsgUpdateVendorInfo_ValidateBasic(t *testing.T) {
 				VendorLandingPageURL: testconstants.VendorLandingPageURL,
 			},
 		},
+		{
+			name: "optional vendor name is not set",
+			msg: MsgUpdateVendorInfo{
+				Creator:              sample.AccAddress(),
+				VendorID:             testconstants.VendorID1,
+				VendorName:           "",
+				CompanyLegalName:     testconstants.CompanyLegalName,
+				CompanyPreferredName: testconstants.CompanyPreferredName,
+				VendorLandingPageURL: testconstants.VendorLandingPageURL,
+			},
+		},
+		{
+			name: "optional company legal name is not set",
```

Comment:
Please add the following tests as well:
- `CompanyPreferredName` is empty
- `VendorLandingPageURL` is empty
- all optional fields are empty